frontend/components/Archives.tsx

```diff
@@ -42,186 +39,232 @@ export interface IFetchArchives {
     startFrom: number,
     amount: number,
     setArchives: React.Dispatch<React.SetStateAction<IArchive[]>>,
-    setLoadedIndices: React.Dispatch<React.SetStateAction<number>>,
+    setLastLoadedIndex: React.Dispatch<React.SetStateAction<number>>,
     ownerProfiles: { [key: string]: any },
     setOwnerProfiles: React.Dispatch<
       React.SetStateAction<{ [key: string]: any }>
     >,
   ): Promise<void>;
 }
 
+export interface IFetchArchivesCount {
+  (
+    setArchivesCount: React.Dispatch<React.SetStateAction<number | undefined>>,
+  ): Promise<void>;
+}
+
 interface IOwners {
   [key: string]: LSP3ProfileMetadata;
 }
 
 interface ArchivesProps {
   fetchArchives: IFetchArchives;
+  fetchArchivesCount: IFetchArchivesCount;
 }
 
-const Archives: React.FC<ArchivesProps> = ({ fetchArchives }) => {
+const Archives: React.FC<ArchivesProps> = ({
+  fetchArchivesCount,
+  fetchArchives,
```

Comment:
I feel we're reaching a point where re-using this Archives component for both archives and contributions is reaching a point where maybe splitting them and having a bit of duplication is better than this state

They have different way to fetch data, and they display different items - and it's hard to follow what the component because the logic is spread around

Initially we only intended to have one carousel but If our aim is to re-use carousel featues, we may his a Carousel component which renders "things" and pass result/data down

Reply:
@democrazyfi realize that things have changed from original, but core logic is pretty straightforward. i can explain on call. feel like reworking this now would take a while. how data is fetched feels like should be outside of carousel component logic anyways. whereas carousel purely takes care of rendering and knowing when to trigger fetching next batch of data